ui/components/FilterBar.tsx

```diff
@@ -0,0 +1,130 @@
+import {
+  Checkbox,
+  List,
+  ListItem,
+  ListItemIcon,
+  Popover,
+} from "@material-ui/core";
+import _ from "lodash";
+import * as React from "react";
+import styled from "styled-components";
+import Button from "./Button";
+import Flex from "./Flex";
+import Icon, { IconType } from "./Icon";
+import Spacer from "./Spacer";
+import Text from "./Text";
+
+const StyledPopover = styled(Popover)`
+  .MuiPopover-paper {
+    min-width: 450px;
+    border-left: 2px solid ${(props) => props.theme.colors.neutral30};
+    padding-left: ${(props) => props.theme.spacing.medium};
+  }
+  .MuiListItem-gutters {
+    padding-left: 0px;
+  }
+  .MuiCheckbox-root {
+    padding: 0px;
+  }
+  .MuiCheckbox-colorSecondary {
+    &.Mui-checked {
+      color: ${(props) => props.theme.colors.primary};
+    }
+  }
+`;
+
+/** Filter Bar Properties */
+export interface Props {
+  className?: string;
+  /** currently checked filter options. Part of a `useState` with `setActiveFilters` */
+  activeFilters: string[];
+  /** the setState function for `activeFilters` */
+  setActiveFilters: (newFilters: string[]) => void;
+  /** Object containing column headers + corresponding filter options */
+  filterList: { [header: string]: string[] };
+}
+
+/** Form Filter Bar */
+function UnstyledFilterBar({
+  className,
+  activeFilters,
+  setActiveFilters,
+  filterList,
+}: Props) {
+  /** why isn't this a ref? It doesn't work. In The MUI Popover docs they do it with setState too... :( https://mui.com/components/popover/ */
+  const [anchorEl, setAnchorEl] = React.useState(null);
+  const [showFilters, setShowFilters] = React.useState(false);
+
+  const onCheck = (e, option) => {
```

Comment:
Can we get some type annotations here? What will `e` and `option` look like when they get passed in here?